numpy/polynomial/polynomial.py

```diff
@@ -1441,27 +1441,31 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None):
 
     Examples
     --------
-    >>> np.random.seed(123)
     >>> from numpy.polynomial import polynomial as P
     >>> x = np.linspace(-1,1,51)  # x "data": [-1, -0.96, ..., 0.96, 1]
-    >>> y = x**3 - x + np.random.randn(len(x))  # x^3 - x + Gaussian noise
+    >>> rng = np.random.default_rng()
+    >>> err = rng.normal(size=len(x))
+    >>> y = x**3 - x + err  # x^3 - x + Gaussian noise
     >>> c, stats = P.polyfit(x,y,3,full=True)
-    >>> np.random.seed(123)
-    >>> c # c[0], c[2] should be approx. 0, c[1] approx. -1, c[3] approx. 1
-    array([ 0.01909725, -1.30598256, -0.00577963,  1.02644286])  # may vary
+    >>> c # c[0], c[1] approx. -1, c[2] should be approx. 0, c[3] approx. 1
+    array([ 0.23111996, -1.02785049, -0.2241444 ,  1.08405657]) # may vary
     >>> stats # note the large SSR, explaining the rather poor results
-     [array([ 38.06116253]), 4, array([ 1.38446749,  1.32119158,  0.50443316, # may vary
-              0.28853036]), 1.1324274851176597e-014]
+    [array([48.312088]),                                        # may vary
+     4,
+     array([1.38446749, 1.32119158, 0.50443316, 0.28853036]),
+     1.1324274851176597e-14]
```

Comment:
I am a bit surprised doctests accepts the formatting change. Is this doctest-ed?

Reply:
It was doctested.  I was a bit surprised too.  I think something is causing it to miss this function. It's on my to-do list to explore why it was skipped.

Reply:
Our doctest runner [normalizes whitespace by default](https://github.com/numpy/numpy/blob/main/tools/refguide_check.py#L803). IPython-pretty-printed lists are acceptable (see my own journey of discovery [on this other issue](https://github.com/numpy/numpy/pull/26538#discussion_r1616134523)).

Reply:
I think the `mayvary` directive is telling doctests to ignore everything. You can run this locally (after `pip install .`) with the verbose flag and then see the doctest output
```
python tools/refguide_check.py --doctests numpy.polynomial -vv > /tmp/output
```

Basically, anything there will be accepted.

Reply:
That too. :smile:

Reply:
@mattip , I checked it yesterday without the `# may vary` tag and things ran fine. Thanks for the link @rkern on your journey, I'll be following that today. 

The issue I found yesterday is that without the `# may vary` tag, If I changed the output (say a 2.999... to a 5.9999), then the test still passed on multiple of the functions in the `numpy polynomial` module.  I was able to break the doctest in other places, but not there.  I have to spend some time to make something reproducible, then I'll start a different issue that references this discussion (or fix the issue once I narrow it down).

Reply:
It was a `user error`. :)  No issue with the doc test.  I followed the workflow in `.circleci/config.yaml`, and found the missing bit.  You mentioned it above, namely `pip install .`

Reply:
> the mayvary directive is telling doctests to ignore everything

As a belated comment, `# may vary` make the doctester ignore the output (https://github.com/scipy/scipy_doctest/blob/main/scipy_doctest/impl.py#L271)

The difference to `# doctest: +SKIP` is two-fold:
- `# doctest: +SKIP` should be added the example (the line starting with `>>>`, in the doctest parlance, it's `example.source`); `# may vary` goes with the output (in the doctest language, `want`)
- `# doctest: +SKIP` skips checking that the example is legal python; `# may vary` does not; so it's really "check that the example.source produces something, and don't bother checking this something"